src/v/storage/tests/compaction_fuzz_test.cc

```diff
@@ -39,7 +37,7 @@ static model::record_batch make_random_batch(
   std::vector<std::optional<ss::sstring>> keys,
   std::vector<std::optional<ss::sstring>> values,
   int num_records) {
-    BOOST_REQUIRE(keys.size() == values.size());
+    assert(keys.size() == values.size());
```

Comment:
nit: wdyt about making a simple gtest fixture that includes this method and uses ASSERT_EQ?

Or at the very least use vassert instead of assert so it logs the line number